### PR TITLE
TST: Streaming of S3 files

### PR DESCRIPTION
**Pull request by kokes**
Issue #17135 was resolved by an upstream change, this is just a unit test to avoid regressions.

### CODE REVIEW

**Comment by kokes** [comment]
CI will crash and burn as I didn't add `botocore` as a dependency properly. Any help on this appreciated.

**Comment by TomAugspurger** [comment]
I think the easiest thing to do is to ensure the test is skipped without botocore, and rely on `s3fs` pulling it in as a dependency.

File: ci/environment-dev.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ dependencies:
   - flake8-comprehensions
   - hypothesis>=3.58.0
   - moto
+  - botocore>=1.10.47
```
Comment (TomAugspurger) [inline ci/environment-dev.yaml]: So this can be removed.

File: pandas/tests/io/test_s3.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
+from six import BytesIO
+
+from botocore.response import StreamingBody
```
Comment (TomAugspurger) [inline pandas/tests/io/test_s3.py]: And remove this.

File: pandas/tests/io/test_s3.py [hunk 3]
```diff
@@ -6,3 +10,13 @@ class TestS3URL(object):
     def test_is_s3_url(self):
         assert is_s3_url("s3://pandas/somethingelse.com")
         assert not is_s3_url("s4://pandas/somethingelse.com")
+
+
+def test_streaming_s3_objects():
+    data = [
```
Comment (TomAugspurger) [inline pandas/tests/io/test_s3.py]: Then here you can do `botocore_response = pytest.importorskip('botocore.response', minversion='1.10.47')`

File: pandas/tests/io/test_s3.py [hunk 4]
```diff
@@ -1,3 +1,7 @@
+from six import BytesIO
```
Comment (TomAugspurger) [inline pandas/tests/io/test_s3.py]: `from pandas.compat import BytesIO`

File: pandas/tests/io/test_s3.py (hunk 3 above)
Comment (kokes) [reply pandas/tests/io/test_s3.py]: Can we do this when the version pertains to the base module, not the imported class? I just tried it in a dummy file and all I got was `Skipped: module 'botocore.response' has __version__ None, required is: '1.10.47'`. So should I do something like this?

```
import pytest

def my_test_fn():
    pytest.importorskip('botocore', minversion='1.10.47')
    from botocore.response import StreamingBody
    # test
```

File: pandas/tests/io/test_s3.py [hunk 5]
```diff
@@ -6,3 +10,16 @@ class TestS3URL(object):
     def test_is_s3_url(self):
         assert is_s3_url("s3://pandas/somethingelse.com")
         assert not is_s3_url("s4://pandas/somethingelse.com")
+
+
+def test_streaming_s3_objects():
+    pytest.importorskip('botocore', minversion='1.10.47')
```
Comment (jreback) [inline pandas/tests/io/test_s3.py]: can you add the issue number here as a comment

**Review by jreback — CHANGES_REQUESTED** [review]
doc comment ping on green.

File: pandas/tests/io/test_s3.py (hunk 5 above)
Comment (kokes) [reply pandas/tests/io/test_s3.py]: yup, done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**